src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs

```diff
@@ -38,6 +45,78 @@ public static void ProcessLogFile(string filePath, Solution solution, string too
             if (toolFormat.MatchesToolFormat(ToolFormat.None))
             {
                 logText = File.ReadAllText(filePath);
+
+                if (logText.Contains(@"""version"": ""1.0.0"""))
+                {
+                    // They're opening a v1 log, so we need to transform it
+                    // Ask if they'd like to save the v2 log
+                    int response = VsShellUtilities.ShowMessageBox(SarifViewerPackage.ServiceProvider,
+                                                                   Resources.TransformV1_DialogMessage,
+                                                                   null, // title
+                                                                   OLEMSGICON.OLEMSGICON_QUERY,
+                                                                   OLEMSGBUTTON.OLEMSGBUTTON_YESNOCANCEL,
+                                                                   OLEMSGDEFBUTTON.OLEMSGDEFBUTTON_FIRST);
+
+                    if (response == IDCANCEL)
+                    {
+                        return;
+                    }
+
+                    JsonSerializerSettings settingsV1 = new JsonSerializerSettings()
+                    {
+                        ContractResolver = SarifContractResolverVersionOne.Instance,
+                    };
+
+                    SarifLogVersionOne v1Log = JsonConvert.DeserializeObject<SarifLogVersionOne>(logText, settingsV1);
+                    var transformer = new SarifVersionOneToCurrentVisitor();
+                    transformer.VisitSarifLogVersionOne(v1Log);
+                    log = transformer.SarifLog;
+
+                    if (response == IDYES)
+                    {
+                        // Prompt for a location to save the transformed log
+                        var saveFileDialog = new SaveFileDialog();
```

Comment:
None of this code is unit testable because it pops UI. Please file an issue to revisit and add tests.